tests/pos/opaque-propability-xm.scala

```diff
@@ -0,0 +1,41 @@
+object prob {
+  opaque type Probability = Double
+
+  implicit object Probability {
+    def apply(n: Double): Option[Probability] =
+      if (0.0 <= n && n <= 1.0) Some(n) else None
+
+    def unsafe(p: Double): Probability = {
+      require(0.0 <= p && p <= 1.0, s"probabilities lie in [0, 1] (got $p)")
+      p
+    }
+
+    def asDouble(p: Probability): Double = p
+
+    val Never: Probability = 0.0
+    val CoinToss: Probability = 0.5
+    val Certain: Probability = 1.0
+
+    implicit val ordering: Ordering[Probability] =
+      implicitly[Ordering[Double]]
+
+    def (p1: Probability) unary_~ : Probability = Certain - p1
+    def (p1: Probability) & (p2: Probability): Probability = p1 * p2
+    def (p1: Probability) | (p2: Probability): Probability = p1 + p2 - (p1 * p2)
```

Comment:
Minor note: these formulas, though they appeared in the proposal, are a bit misleading. They are true of indendent events but not all events. Below you use them with missedTrain and caughtTrain. They are not indendent since you can’t miss and catch the train.

The real probability of missedTrain | caughtTrain is 1 but this formula won’t show that.

To fix the example it might be easiest to choose three events one imagines might be independent.

It doesn’t matter much but it might be nicer to avoid a future “well actually” (which technically this is, but I’m only doing it to give us a chance to possibly change it before some jerk posts to Twitter about how scala programmers don’t understand probability).